javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentClassLoader.java

```diff
@@ -158,7 +167,14 @@ public Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundExce
       }
       // search from parent and urls added to this loader
       if (clazz == null) {
-        clazz = super.loadClass(name, false);
+        try {
```

Comment:
WDYT about wrapping `InstrumentationModuleClassLoader.defineClassWithPackage` in
```java
try {
  ...
} catch (LinkageError error) {
  Class<?> clazz = findLoadedClass(name);
  if (clazz != null) {
    return clazz;
  }
  throw error;
}
```
Since we can't exactly control when the load class instrumentation is applied and what is called right after it it is hard to reason whether this the same issue could happen in some other way.

Reply:
Implemented in 6662850